File: packages/react-native-fast-ws/android/src/main/java/com/margelo/nitro/websocket/HybridWebSocketManager.kt
package com.margelo.nitro.websocket

import com.facebook.proguard.annotations.DoNotStrip

@DoNotStrip
class HybridWebSocketManager : HybridWebSocketManagerSpec() {
    override fun create(url: String, protocols: Array<String>): HybridWebSocketSpec {
        return HybridWebSocket(url, protocols)
    }

    override val memorySize: Long
        get() = 0L
}

File: packages/react-native-fast-io/ios/HybridWebSocketManager.swift
//
//  HybridStyleSheet.swift
//  ReactNativeFastWS
//
//  Created by Mike Grabowski on 13/09/2024.
//

import NitroModules

class HybridWebSocketManager : HybridWebSocketManagerSpec {
  func create(url: String, protocols: [String]) throws -> any HybridWebSocketSpec {
    return HybridWebSocket(url: url, protocols: protocols)
  }
}
